server/rooms.js

```diff
@@ -638,12 +638,17 @@ class GlobalRoom extends BasicRoom {
 			if (room.type !== 'battle') continue;
 			if (formatFilter && formatFilter !== room.format) continue;
 			if (eloFilter && (!room.rated || room.rated < eloFilter)) continue;
+			const p1userid = room.battle && room.battle.p1.userid;
+			const p2userid = room.battle && room.battle.p2.userid;
+			if (usernameFilter && ((p1userid && !p1userid.startsWith(usernameFilter)) && (p2userid && !p2userid.startsWith(usernameFilter)))) {
```

Comment:
confused by this conditional - isnt this saying both usernames need to start with the username filter?

Reply:
No, it will only skip the iteration if both don't start with the prefix.

Reply:
That conditional will only be `true` if both of them don't start with the prefix, if we used `||` it could be `true` even if one of them starts with the prefix. `true` in this case means 'skip' this battle.

Reply:
Maybe something like this would be more clear?

```ts
if (userfilter && room.battle) {
  if (!room.battle.p1.userid || !room.battle.p2.userid) continue;
  if (!(room.battle.p1.userid.startsWith(usernameFilter) || !room.battle.p2.userid.startsWith(usernameFilter)) continue;
}
```